pdsfile.py

```diff
@@ -1305,10 +1305,12 @@ def os_listdir(abspath):
                     # If the shelf file is missing, try the actual file system
                     # For documentation, we have all files available but not the shelf
                     # files, therefore we will check the actual file system for documents.
-                    childnames = os.listdir(abspath)
-                    return [c for c in childnames
-                            if c != '.DS_Store' and not c.startswith('._')]
-
+                    if '/holdings/documents' in abspath:
+                        childnames = os.listdir(abspath)
+                        return [c for c in childnames
+                                if c != '.DS_Store' and not c.startswith('._')]
+                    else:
+                        return []
                 if not results:
```

Comment:
This change does not look correct to me. Previously, if a shelf file was missing, it would check the file system. Now, it only checks the file system if the file is in the documents subdirectory. I don't know why we would want this restriction. It probably didn't break anything in the current system, but just because we are careful about managing our shelf files, but I recommend you revert this change unless it serves some purpose I don't see.

Reply:
The change is reverted.